### PR TITLE
python3Packages.typish: init at 1.9.3

### PR DESCRIPTION
**Pull request by fmoda3**
###### Description of changes

Add [typish](https://pypi.org/project/typish/), a library for checking types of python objects

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -4199,6 +4199,12 @@
     githubId = 74379;
     name = "Florian Pester";
   };
+  fmoda3 = {
+    email = "fmoda3@mac.com";
+    github = "fmoda3";
+    githubId = 1746471;
+    name = "Frank Moda III";
+  };
```
Comment (pbsds) [inline maintainers/maintainer-list.nix:4207]: Please separate this into a separate commit, titled `maintainers: add fmoda3`.

**Review by pbsds — CHANGES_REQUESTED** [review]
Looking good. Seems to work. Only a minor rebase needed to fit https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (jonringer) [inline maintainers/maintainer-list.nix:4207]: Please make this into a separate commit:

```
maintainers: add <name>
```

File: pkgs/development/python-modules/typish/default.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+}:
+
+buildPythonPackage rec {
+  pname = "typish";
+  version = "1.9.3";
+  format = "wheel";
```
Comment (jonringer) [inline pkgs/development/python-modules/typish/default.nix:9]: wheels are discouraged as they likely lack tests, if possible, please try to package this using source and ensure that tests are being ran

File: maintainers/maintainer-list.nix (hunk 1 above)
Comment (fmoda3) [reply maintainers/maintainer-list.nix:4412]: Pulled this out into a separate commit

Comment (fmoda3) [reply maintainers/maintainer-list.nix:4412]: Pulled this out into a separate commit

File: pkgs/development/python-modules/typish/default.nix (hunk 2 above)
Comment (fmoda3) [reply pkgs/development/python-modules/typish/default.nix]: Switched it to pull from source and run tests.  One of the tests unfortunately depends on a super old version of `nptyping` which appears to circular depend back on `typish`.  All the other tests pass.

**Comment by pbsds** [comment]
Result of `nixpkgs-review pr 167795` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.typish</li>
    <li>python39Packages.typish</li>
  </ul>
</details>

**Review by pbsds — APPROVED** [review]

**Review by viraptor — APPROVED** [review]
Result of `nixpkgs-review pr 167795` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python310Packages.typish</li>
    <li>python39Packages.typish</li>
  </ul>
</details>
